ARCHITECTURE.md

```diff
@@ -170,3 +188,49 @@ one point.
 
 The prefix `seal` here is for historic reasons. There is some analogy to sealing a
 contract. And we found seals to be a cute animal as well ‒ like squids!
+
+## `Environment` Trait
+
+Our `ink_env` crate defines a trait [`Environment`](https://paritytech.github.io/ink/ink_env/trait.Environment.html).
+It also exposes a [`DefaultEnvironment`](https://paritytech.github.io/ink/ink_env/enum.DefaultEnvironment.html):
+
+```rust
+/// The fundamental types of the default configuration.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(TypeInfo))]
+pub enum DefaultEnvironment {}
+
+impl Environment for DefaultEnvironment {
+    const MAX_EVENT_TOPICS: usize = 4;
+
+    type AccountId = ink_primitives::AccountId;
+    type Balance = u128;
+    type Hash = ink_primitives::Hash;
+    type Timestamp = u64;
+    type BlockNumber = u32;
+    type ChainExtension = NoChainExtension;
+}
+```
```

Comment:
Too low-level for the `ARCHITECTURE.md` file.

```suggestion
```

This should instead be part of the `ink_env` crate docs